app/services/state_file/messaging_service.rb

```diff
@@ -1,26 +1,69 @@
 module StateFile
   class MessagingService
-    def self.send_email(intake:, subject:, body:)
-      raise ArgumentError unless intake.present? && subject.present? && body.present?
-      return unless intake.email_address.present? && intake.email_address_verified_at.present?
-
-      StateFileNotificationEmail.create!(
-        to: intake.email_address,
-        body: body,
-        subject: subject,
+    attr_accessor :locale, :message, :intake, :locale, :message_tracker, :sent_messages
+
+    def initialize(message:, intake:, locale: nil, sms: true, email: true, body_args: {})
+      @locale = locale || "en" #|| intake.locale || "en", add the intake locale once start collectin
+      @message = message
+      @message_instance = message.new
+      @intake = intake
+      @sent_messages = []
+      @body_args = body_args
+      @do_sms = sms
+      @do_email = email
+    end
+
+    def message_tracker
+      @message_tracker ||= MessageTracker.new(data_source: intake, message: message)
+    end
+
+    def send_message
+      return nil if message_tracker.already_sent? && message.send_only_once?
+
+      send_email if @do_email
+      # send_sms if @do_sms # TODO: Eventually send SMS when fixed
+
+      message_tracker.record(sent_messages.last.created_at) if sent_messages.any? # will this be recorded correctly with what we have on line 40
+
+      sent_messages
+    end
+
+    private
+
+    def send_email
+      return unless Flipper.enabled?(:state_file_notification_emails) && intake.email_address.present? && intake.email_address_verified_at.present?
+
+      if @message_instance.email_body.present?
+        sent_message = StateFileNotificationEmail.create!(
+          data_source: intake,
+          to: intake.email_address,
+          body: @message_instance.email_body(locale: locale, **email_args),
+          subject: @message_instance.email_subject(locale: @locale, **email_args),
         )
+        sent_messages << sent_message if sent_message.present?
+      end
     end
+    
+    # def send_sms
+    #   return unless Flipper.enabled?(:sms_notifications)
+    # end
 
-    def self.send_notification(intake:, message:, body_args: {})
-      # TODO: Figure out locale from intake somehow (though we don't currently capture preferred language anywhere)
-      locale = nil
+    def base_args
+      first_name = intake.primary_first_name ? intake.primary_first_name.split(/ |\_/).map(&:capitalize).join(" ") : ""
```

Comment:
@embarnard I don't understand this part, isn't primary_first_name already a first name that is provided during intake?

Reply:
this is just for formatting what the client will see in their message. Many of them start with "Hi Name-of-client" but if we save primary_first_name as "monA lISa" we want it to display in the message as "Mona Lisa"